### ISSUE TITLE
"astro dev restart" not working

### ISSUE THREAD

**Issue by casper-voltgoed** [astronomer/astro-cli#1252]
**Describe the bug**

When I run the command `astro dev restart` after running  `astro dev start` the containers are stopped but afterwards an error ocmes up saying it cannot start, so i have to manually run `astro dev start` to start again. See the full output below:

```bash
> astro dev restart
Env file ".env" found. Loading...
[+] Running 4/4
 - Container ***-astro-airflow_ad862e-triggerer-1  Stopped                                                                                                            3.4s 
 - Container ***-astro-airflow_ad862e-webserver-1  Stopped                                                                                                            9.8s 
 - Container ***-astro-airflow_ad862e-scheduler-1  Stopped                                                                                                            3.7s
 - Container ***-astro-airflow_ad862e-postgres-1   Stopped                                                                                                            0.5s
Error: cannot start, project already running
>
```

**What CLI Version did you experience this bug?**

Astro CLI Version: 1.14.1

**This CLI bug is related to which Astronomer Platform?**

- [x] Astro
- [ ] Software
- [ ] None/Unknown

**What Operating System is the above CLI installed on?**

Windows 11

**🪜 Steps To Reproduce**

1. Run `astro dev start`, wait till the project is fully running.
2. Run `astro dev restart`

**Comment by neel-astro** [comment]
Tried it locally with latest Astro CLI version, and `astro dev restart` seems to be working fine:
```
neel@Neels-MacBook-Pro airflow-test-15 % ../astro dev restart
Env file ".env" found. Loading...
[+] Running 4/4
 ⠿ Container airflow-test-15_9710b1-webserver-1  Stopped                                                                                                              0.5s
 ⠿ Container airflow-test-15_9710b1-triggerer-1  Stopped                                                                                                              1.6s
 ⠿ Container airflow-test-15_9710b1-scheduler-1  Stopped                                                                                                              1.9s
 ⠿ Container airflow-test-15_9710b1-postgres-1   Stopped                                                                                                              0.1s
[+] Building 0.3s (11/11) FINISHED                                                                                                                                         
 => [internal] load build definition from Dockerfile                                                                                                                  0.0s
 => => transferring dockerfile: 86B                                                                                                                                   0.0s
 => [internal] load .dockerignore                                                                                                                                     0.0s
 => => transferring context: 85B                                                                                                                                      0.0s
 => [internal] load metadata for quay.io/astronomer/astro-runtime:8.6.0                                                                                               0.3s
 => [1/1] FROM quay.io/astronomer/astro-runtime:8.6.0@sha256:4492f837dde4a6a16fa55fd7e3abaae3b66de5cbe7a8ab2ce07d9e57e5577449                                         0.0s
 => [internal] load build context                                                                                                                                     0.0s
 => => transferring context: 1.27kB                                                                                                                                   0.0s
 => CACHED [2/1] COPY packages.txt .                                                                                                                                  0.0s
 => CACHED [3/1] RUN if [[ -s packages.txt ]]; then     apt-get update && cat packages.txt | tr '\r\n' '\n' | sed -e 's/#.*//' | xargs apt-get install -y --no-insta  0.0s
 => CACHED [4/1] COPY requirements.txt .                                                                                                                              0.0s
 => CACHED [5/1] RUN if grep -Eqx 'apache-airflow\s*[=~>]{1,2}.*' requirements.txt; then     echo >&2 "Do not upgrade by specifying 'apache-airflow' in your require  0.0s
 => CACHED [6/1] COPY --chown=astro:0 . .                                                                                                                             0.0s
 => exporting to image                                                                                                                                                0.0s
 => => exporting layers                                                                                                                                               0.0s
 => => writing image sha256:cc12acd69295d595baf089ac0c269f0f34cb1f013a1ca68d8757f8ccaaf4de6b                                                                          0.0s
 => => naming to docker.io/airflow-test-15_9710b1/airflow:latest                                                                                                      0.0s
Env file ".env" found. Loading...
[+] Running 4/4
 ⠿ Container airflow-test-15_9710b1-postgres-1   Started                                                                                                              0.2s
 ⠿ Container airflow-test-15_9710b1-scheduler-1  Started                                                                                                              1.1s
 ⠿ Container airflow-test-15_9710b1-webserver-1  Started                                                                                                              1.8s
 ⠿ Container airflow-test-15_9710b1-triggerer-1  Started                                                                                                              1.0s


Airflow is starting up!
airflow pools set  default_pool 128 'Default pool' 
Added Pool: default_pool

Project is running! All components are now available.

Airflow Webserver: http://localhost:8090
Postgres Database: localhost:5455/postgres
The default Airflow UI credentials are: admin:admin
The default Postgres DB credentials are: postgres:postgres
```

@casper-voltgoed  Could you retry with the latest Astro CLI version, also is this issue happening all the time, or an intermittent thing?

**Comment by casper-voltgoed** [comment]
@neel-astro I tried the newest version, 1.16.2, and it still occurs, I got the exact same output as in my original description.

It happens every time for me.

**Comment by neel-astro** [comment]
Hmm, that's odd. @casper-voltgoed Can you run a local dev project via `astro dev start`, and then run `astro dev stop` to stop the project and then share the output of `astro dev ps` command

**Comment by casper-voltgoed** [comment]
@neel-astro , Here is the full output of running `astro dev stop` and `astro dev ps` afterwards:
```
> astro dev stop
[+] Running 4/4
 - Container ***-astro-airflow_ad862e-webserver-1  Stopped                                                                                                                                                                                             8.2s 
 - Container ***-astro-airflow_ad862e-triggerer-1  Stopped                                                                                                                                                                                             4.0s 
 - Container ***-astro-airflow_ad862e-scheduler-1  Stopped                                                                                                                                                                                             3.3s
 - Container ***-astro-airflow_ad862e-postgres-1   Stopped                                                                                                                                                                                             0.5s
> astro dev ps
Name                                            State   Ports
***-astro-airflow_ad862e-webserver-1       exited
***-astro-airflow_ad862e-scheduler-1       exited
***-astro-airflow_ad862e-triggerer-1       exited
***-astro-airflow_ad862e-postgres-1        exited
>
```

**Comment by neel-astro** [comment]
@casper-voltgoed the containers seem to be in the correct state once stopped. Can you validate if there is some delay between `astro dev stop` stopping the containers and containers actually being in the exited state on your local machine since this seems to be an issue scoped to your local machine only as of now, so could be either linked to your local docker settings or workload you are running as part of your local setup?

I can't think of any other reason why you are observing this behavior, since `astro dev restart` is equivalent to `astro dev stop && astro dev start`, and the only reason why `astro dev restart` would fail to start containers, but `astro dev start` works fine otherwise on your local machine could be because of additional delay that manually executing `astro dev start` gets over `astro dev restart`

**Comment by cdabella** [comment]
@neel-astro I've been fighting the same thing with my CLI as well, I've got a recording I'll share with you. It looks like it's the postgres container. The start is called before it's finished shutting down. Screenshot included.

![image](https://github.com/astronomer/astro-cli/assets/5761024/c9013221-266e-4645-a63d-0c99845c1b4d)


Astro CLI Version: 1.16.1
Windows 10
WSL

**Comment by neel-astro** [comment]
This has been fixed in the upcoming 1.18 CLI release, planned to be released this week. Closing this ticket for now, feel free to reopen it if you guys still face the issue with the 1.18.0 CLI release.
